### ISSUE TITLE
ADD MULTIMODAL SUPPORT FOR AWESOME TIME SERIES FUSION!!! 🚀🔥

### ISSUE THREAD

**Issue by WhisperBamb** [kwuking/TimeMixer#153]
Hey TimeMixer team! I've been thinking - wouldn't it be AMAZING if we could enhance the model with **multimodal data fusion capabilities**?!? Many real-world time series problems have complementary data modalities that could boost performance!

### THE VISION ✨
Let's extend TimeMixer to accept and process multiple data types alongside the primary time series, such as:
- 📊 Tabular static features (patient demographics, asset characteristics)
- 📝 Text descriptions (event logs, reports)
- 🖼️ Image data (satellite imagery for weather forecasting, medical images)

### WHAT I'M THINKING 💭
A simple first implementation could focus on tabular+time series fusion:

```python
class MultimodalTimeMixer(nn.Module):
    def __init__(self, time_config, tabular_config):
        super().__init__()
        
        # Original TimeMixer for temporal data
        self.time_mixer = TimeMixer(time_config)
        
        # MLP for tabular features
        self.tabular_net = nn.Sequential(
            nn.Linear(tabular_config.in_dim, tabular_config.hidden_dim),
            nn.ReLU(),
            nn.Linear(tabular_config.hidden_dim, tabular_config.out_dim)
        )
        
        # Fusion mechanism
        self.fusion = nn.Linear(
            time_config.d_model + tabular_config.out_dim,
            time_config.d_model
        )
    
    def forward(self, time_data, tabular_data, *args):
        # Get temporal embeddings
        time_out = self.time_mixer.encode(time_data, *args)
        
        # Get tabular embeddings
        tab_out = self.tabular_net(tabular_data)
        
        # Fuse representations
        if len(tab_out.shape) == 2:  # [B, D]
            tab_out = tab_out.unsqueeze(1).expand(-1, time_out.shape[1], -1)
            
        fused = torch.cat([time_out, tab_out], dim=-1)
        fused = self.fusion(fused)
        
        # Decode as usual
        return self.time_mixer.decode(fused, *args)
```

### BENEFITS!!! 🌟
1. **Better accuracy** - leverage ALL available data sources!
2. **More applications** - solve problems that need context beyond just the time series
3. **Competitive advantage** - other forecasting models don't handle this well!

Would LOVE to help implement this if you're interested! What do you think? 😃
